libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -32,9 +36,14 @@ export class BookSearchComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.store.select(getAllBooks).subscribe(books => {
+    const readingListBookSubscription = this.store.select(getAllBooks).subscribe(books => {
       this.books = books;
     });
+    this.subscription.add(readingListBookSubscription);
+  }
+
+  ngAfterViewInit(): void {
```

Comment:
Why did you choose to use this lifecycle method? Is it really needed? Why yes, why no?

Reply:
Why did you choose to use this lifecycle method? Is it really needed?
my answer: I would say that it depends on whether or not this code is deemed an initialization task or not. Since I was in between I chose afterViewInit simply because this code listens for user changes that happen after the view is initialized and felt this LCE was more appropriate. An argument could be made that AfterViewInit is ideal for handling tasks that requires/ensures that its child components are initialized before executing which isn't the case here. I don't have any strong thoughts here as I am not well versed in angular and would defer to you & those with more experience in angular.  if you would like to expand on the angular documentation I'm all ears.

Reply:
Thank you for the explanation. Indeed, the afterViewInit is usually used when you want to make sure that all view childs are there. In any case, we can leave this there now, but I would suggest putting it in ngOnInit next time!